lib/mongo/cluster.rb

```diff
@@ -359,8 +363,8 @@ def servers
     #
     # @since 2.1.0
     def disconnect!
```

Comment:
Is this called by the Ruby garbage collector? If so I'd like to discuss whether a deadlock is possible akin to PYTHON-799. If this isn't called during garbage collection, nevermind. =)